src/menu/menu.c

```diff
@@ -1625,7 +1626,8 @@ menu_execute_item(struct menuitem *item)
 	struct server *server = item->parent->server;
 	menu_close(server->menu_current);
 	server->input_mode = LAB_INPUT_STATE_PASSTHROUGH;
-	cursor_update_focus(server);
+	seat_finish_server_action(&server->seat,
```

Comment:
Should we consider setting `server->input_mode` in `seat_{begin,finish}_server_action()` functions? They might be set in other places - not looked - but it would feel like a neat thing to do.

Reply:
Yeah you're absolutely right, and that's what I described as TODO in the header file. As I described there, I think the cleanest option is to add `LAB_INPUT_STATE_WINDOW_SWITCHER` in the `input_mode` enum and set it in those functions, but I didn't in this PR since it's kind of unpredictable how it affects other places like `input_mode != LAB_INPUT_STATE_PASSHROUGH` checks.

I'm thinking of opening another PR for just adding `LAB_INPUT_STATE_WINDOW_SWITCHER`.

Reply:
Done, with help of #2459.